edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/ModuleLogsRequestHandler.cs

```diff
@@ -50,23 +52,29 @@ protected override async Task<Option<IEnumerable<ModuleLogsResponse>>> HandleReq
             IEnumerable<Task<ModuleLogsResponse>> uploadLogsTasks = logOptionsList.Select(
                 async l =>
                 {
-                    Events.ReceivedLogOptions(l);
-                    ModuleLogOptions logOptions = l.logOptions.Filter.Tail
-                        .Filter(t => t < MaxTailValue)
-                        .Map(t => l.logOptions)
-                        .GetOrElse(
-                            () =>
-                            {
-                                var filter = new ModuleLogFilter(Option.Some(MaxTailValue), l.logOptions.Filter.Since, l.logOptions.Filter.Until, l.logOptions.Filter.LogLevel, l.logOptions.Filter.RegexString);
-                                return new ModuleLogOptions(l.logOptions.ContentEncoding, l.logOptions.ContentType, filter, l.logOptions.OutputFraming, l.logOptions.OutputGroupingConfig, l.logOptions.Follow);
-                            });
-
-                    byte[] moduleLogs = await this.logsProvider.GetLogs(l.id, logOptions, cancellationToken);
+                    byte[] moduleLogs = await this.logsProvider.GetLogs(l.id, l.logOptions, cancellationToken);
 
                     Events.ReceivedModuleLogs(moduleLogs, l.id);
-                    return logOptions.ContentEncoding == LogsContentEncoding.Gzip
-                        ? new ModuleLogsResponse(l.id, moduleLogs)
-                        : new ModuleLogsResponse(l.id, moduleLogs.FromBytes());
+
+                    if (l.logOptions.ContentEncoding == LogsContentEncoding.Gzip)
+                    {
+                        if (moduleLogs.Length > MaxPayloadSize)
+                        {
+                            return new ModuleLogsResponse(l.id, Events.LargePayload(moduleLogs.Length, l.logOptions));
```

Comment:
This non-trivial change to the behavior of the product code should probably be its own PR.